compiler/qsc_ast/src/ast.rs

```diff
@@ -874,7 +874,7 @@ pub enum ExprKind {
     Err,
     /// A failure: `fail "message"`.
     Fail(Box<Expr>),
-    /// A field accessor: `a::F`.
+    /// A field accessor: `a::F` or `a.F`.
```

Comment:
is this really true? I thought at the AST level the new dot-based field accessors all show up as `Path`s in this enum.

Reply:
When the leading term of a `.` sequence can be parsed as an expression, it can be differentiated early as a field accessor, otherwise it must be parsed as a path and differentiated during name res. The alternative to this is to change the path struct to allow for a leading expr. I tried that setup before, but it added way more complexity in dealing with paths, so differentiating early for expression-parsable field accessors this ended up being much simpler.

Reply:
Oh, I see. Thanks for clarifying!

Reply:
I can see that this is breaking the tests for the qsharp code gen. An alternative is that I could add a new AST node for this specific case (`StructField`?) and differentiate the code gen logic. That seems a bit overkill though, and I do like the idea that the code gen transforms the use of `::` into `.`. But maybe that is strictly incorrect behavior for the code gen? @idavis for more context.

Reply:
My two cents (after thinking about it some more): AST is supposed to, as much as reasonable, map to the syntax the user typed, however you also have upcoming work to warn against use of `::` for field access. Given those two facts, the "right" syntax is `.` and the qsharp generation logic should treat field access as `.`, effectively cleaning up the deprecated-but-still-supported-for-backcompat syntax of `::`. So rather than updating the parser and the AST to differentiate, the codegen tests should be updated to expect `.`.

Reply:
I changed out the `::` for the `.` in the tests.